pkg/api/login.go

```diff
@@ -329,7 +328,11 @@ func (hs *HTTPServer) redirectURLWithErrorCookie(c *contextmodel.ReqContext, err
 }
 
 func (hs *HTTPServer) samlEnabled() bool {
-	return hs.SettingsProvider.KeyValue("auth.saml", "enabled").MustBool(false) && hs.License.FeatureEnabled(social.SAMLProviderName)
+	if hs.authnService == nil {
+		return false
+	}
```

Comment:
if this because of tests?

Reply:
Yes. I didn't want to add another test for this internal function.

Reply:
I added to the test `authntest.FakeService` and remove this conditional